### ISSUE TITLE
Cost is ignored for MacOS

### ISSUE THREAD

**Issue by kevinoneill** [kean/Nuke#236]
The cost function is ignored for MacOS always returning 1 for any image.

```
    func cost(for image: Image) -> Int {
        #if !os(macOS)
        let dataCost = ImagePipeline.Configuration.isAnimatedImageDataEnabled ? (image.animatedImageData?.count ?? 0) : 0

        // bytesPerRow * height gives a rough estimation of how much memory
        // image uses in bytes. In practice this algorithm combined with a
        // concervative default cost limit works OK.
        guard let cgImage = image.cgImage else {
            return 1 + dataCost
        }
        return cgImage.bytesPerRow * cgImage.height + dataCost

        #else
        return 1
        #endif
    }
```

It should be possible to interrogate image representations to determine the approximate size.

**Comment by kean** [comment]
The [ticket](https://trello.com/c/RXJadvVV) to implement it is in Backlog. I moved it to Nuke 8.1. I'm planning to do some other macOS support improvements as well, but  probably after we get some updates regarding Marzipan. If `UIImage` becomes available on macOS, it would be obsolete.

**Comment by kean** [comment]
[Implemented](https://github.com/kean/Nuke/commit/82c4760ec2e9ccc0096a2317e443c60d304876e1), this change will ship in Nuke 8 Beta 2.

**Comment by kean** [comment]
Shipped in [Nuke 8.0-rc.1](https://github.com/kean/Nuke/releases/tag/8.0-rc.1)
